### PR TITLE
DEV-8748 outlay bar

### PR DESCRIPTION
**Pull request by nick-torres**
**High level description:**

Add outlay bar 
**Technical details:**

Quite a few, basically, I added a check to see whether the RoundedToggle was checked or not. If it is, then we construct the statusoffundschart differently. We don't overlap the two bars, we put one over the top of the other, change the fill from what it was originally (including the tbr bar being patterned fill). There were lots of changes to various sizes to make things line up properly. Also, adjusted the model so that the outlays are pulled from the api response.

**JIRA Ticket:**
[DEV-8748](https://federal-spending-transparency.atlassian.net/browse/DEV-8748)

**Mockup:**
https://app.zeplin.io/project/61f806d9ad825610f7b1fc0a/screen/6282cab43594b212a7cdb67a

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [ ] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [ ] Verified cross-browser compatibility: Chrome, Safari, Firefox, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [ ] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [ ] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [ ] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [ ] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [ ] Design review complete `if applicable`
- [ ] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

File: src/js/components/agency/visualizations/StatusOfFundsChart.jsx [hunk 1]
```diff
@@ -83,7 +85,7 @@ const StatusOfFundsChart = ({
         }, 50);
         window.addEventListener('resize', handleResize);
         return () => window.removeEventListener('resize', handleResize);
-    }, []);
```
Comment (anjenkin) [inline src/js/components/agency/visualizations/StatusOfFundsChart.jsx:86]: the window listener could be in a separate useeffect so we not adding and re-adding every time the window width changes.

File: src/js/components/agency/visualizations/StatusOfFundsChart.jsx [hunk 2]
```diff
@@ -188,242 +200,492 @@ const StatusOfFundsChart = ({
...
-
-        // shift x axis labels to match mock
-        const tickTexts = d3.selectAll(".tick text");
-        tickTexts.each(function mobileTextCount(d, i, n) {
-            if (i === n.length - 1) d3.select(this).attr('dx', '-1.2em');
-            if (isMobile) {
-                if (i === 0) d3.select(this).attr('dx', '0.2em');
+            // extract sorted agency names
+            for (let i = 0; i < sortedNums.length; i++) {
+                // resultNames = resultNames.concat(sortedNums[i].name.split(',')[0]);
```
Comment (anjenkin) [inline src/js/components/agency/visualizations/StatusOfFundsChart.jsx:240]: could we remove this commented out line?

Comment (anjenkin) [inline src/js/components/agency/visualizations/StatusOfFundsChart.jsx:240]: could we remove this commented out line?

**Review by anjenkin — CHANGES_REQUESTED** [review]
Just a couple minor comments. Otherwise, looks great

File: src/js/components/agency/visualizations/StatusOfFundsChart.jsx (hunk 1 above)
Comment (nick-torres) [reply src/js/components/agency/visualizations/StatusOfFundsChart.jsx]: That change wasn't on purpose

**Review by anjenkin — DISMISSED** [review]

**Review by anjenkin — APPROVED** [review]

**Pull request merged**